src/LLVM/Typed.hs

```diff
@@ -84,15 +86,19 @@ instance Typed C.Constant where
 getElementPtrType :: Type -> [C.Constant] -> Type
 getElementPtrType ty cons = ptr i8 -- XXX
 
+getElementType :: Type -> Type
```

Comment:
I think it would be better if this would be a partial function that ensures that only works if you are passing a `PointerType` ensuring that you don’t accidentally use it on the wrong type.